pkg/yamlmeta/map_key_overrides.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package yamlmeta
+
+import "sort"
+
+type overrideMapKeys struct{}
+
+// Visit if `node` is a Map, among its MapItem's that have duplicate keys, removes all but the last.
+// This visitor never returns an non-nil error.
+func (r *overrideMapKeys) visit(node Node) error {
+	mapNode, isMap := node.(*Map)
+	if !isMap {
+		return nil
+	}
+
+	keyToIdxs := r.collectIndexesByKey(mapNode)
+	idxsToRemove := r.selectRedundantIdxs(keyToIdxs)
+	for _, idx := range idxsToRemove {
+		mapNode.Items = append(mapNode.Items[:idx], mapNode.Items[idx+1:]...)
+	}
```

Comment:
how about something like this (havent actually ran):

```
lastItemIndex := map[interface{}]int{}
for idx, item := range mapNode.Items {
	lastItemIndex[item.Key] = idx
}

newItems := []*MapItem{}
seenIndex := map[interface{}]struct{}{}
for _, item := range mapNode.Items {
	idx, found := lastItemIndex[item.Key]
	if !found {
		panic("Internal inconsistency: ...")
	}
	if _, found := seenIndex[item.Key]; !found {
		seenIndex[item.Key] = struct{}{}
		newItems = append(newItems, mapNode.Items[idx])
	}
}

mapNode.Items = newItems
```

Reply:
nice. 👍🏻   Far far simpler, no sorting, etc. And summarizing it sounds a lot like what the desired behavior is: keep only the last.

Reply:
> And summarizing it sounds a lot like what the desired behavior is: keep only the last.

well keep only the last in the "space" of the first one -- because its effectively replacing it (vs delete+insert)